scripts/build_shapes.py

```diff
@@ -80,7 +81,64 @@ def download_GADM(country_code, update=False, out_logging=False):
     return GADM_inputfile_gpkg, GADM_filename
 
 
-def get_GADM_layer(country_list, layer_id, geo_crs, update=False, outlogging=False):
+def replace_nonstd_codes(row, col, country_code, keep_cond=True):
+    if row["GID_0"] != country_code:
+        return country_name_2_two_digits(row["COUNTRY"])
+    else:
+        return row["GID_0"]
+
+
+def filter_gadm(geodf, layer, cc, output_nonstd_to_csv=True, keep_all_areas=True):
+
+    # convert country name representation of the main country (GID_0 column)
+    geodf["GID_0"] = geodf["GID_0"].map(three_2_two_digits_country)
+
+    if keep_all_areas:
+        # in case GID_0 have any exotic values, "COUNTRY" column may be used instead
+        geodf["GID_0"] = geodf.apply(
+            lambda x: replace_nonstd_codes(x, col="GID_0", country_code=cc), axis=1
+        )
+        # "not found" hardcoded according to country_converter conventions
+        geodf.drop(geodf[geodf["GID_0"] == "not found"].index, inplace=True)
+    else:
+        geodf_non_std = geodf[geodf["GID_0"] != cc]
+        geodf.drop(geodf[geodf["GID_0"] != cc].index, inplace=True)
+        logger.warning("Regions with non-standard codes dropped:{geodf_non_std}")
+
+    # country shape should have a single geomerty
+    if (layer == 0) & (geodf.shape[0] > 1):
+        # take the first row only
+        geodf_union = geodf.iloc[[0]]
+        geodf_union = geodf_union.set_geometry(
+            # GeoSeries transformation is neded as Polygon type is not hashable
+            gpd.GeoSeries(unary_union(geodf["geometry"]))  # ,
+            # drop=True,
+            # inplace=True
+        )
+        geodf = geodf_union
```

Comment:
May you provide some examples of countries where multiple rows are found here for layer=0?

Eventually, these lines may be fused as
```
geodf = geodf.iloc[[0]].set_geometry([geodf.unary_union])
```
To be tested, but it should work

For compatibility purposes, I think it may be better to return a dataframe anyway.

Moreover, a warning needs to be added to advise that this is happening

Reply:
I'm testing on data for Pakistan where there are non-standard data both for layers 0 and 1. The output was GeoDataFrame, transformation to GeoSeries was needed to be able feed the geometry column with the new data due to some hashable magic. However, the resulting block is not really elegant :/

The solution you suggest is working brilliantly. Thanks a lot!

Regarding a warning not sure I get the point: do you mean to issue a warning when non-standard areas are being merged? Is it relevant for a user? Probably it would be a good idea to inform that the non-standard codes are present in the data.

Would be grateful on your opinion in this regard

Reply:
I think that the user shall be notified that we are merging the shapes as they may not be with our standard approach.
Anyway, there is no need to notify the nonstandard ids, but just notifying like "contended areas have been found. Shapes are being merged"

Probably, the contended_areas option may be handy. In the future we may apply the same approach for these areas: with set_by country we merge them and with drop we drop them

Reply:
Agree. Warnings revised

What I'm not sure about is how will `unary_union` process countries which consist of multiple connected regions. That seems to be a good idea to track it